.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,27 @@
+Thank you for contributing to JanusGraph.
+
+In order to streamline the review of the contribution we ask you
+to ensure the following steps have been taken:
+
+### For all changes:
+- [ ] Is there an issue associated with this PR? Is it referenced 
+     in the commit message?
+
+- [ ] Does your PR title ends with #xyz where xyz is the issue number you are trying to resolve?
```

Comment:
Hmm, I looked at one of my private repositories and didn't see what you describe here. 

If the issue is in the title/subject line it is shown in the commit history only once. If it is in the body the title doesn't contain the issue number. We have to be careful, this speaks about the git commit message not the title of the PR on Github.

The issue number in the title is also click-able and linked.